### PR TITLE
Add Rainbow beds

### PR DESCRIPTION
**Pull request by svr333**
## Description
Added Rainbow Beds to the Rainbow Blocks collection.
The beds are still spazzing out right now but I havent found the issue with it, though I wanted to push the code already.
Spazzing bed

## Changes
- Create Rainbow Bed SlimefunItem
- Add Rainbow Bed to SlimefunItemSetup
- Add Rainbow Bed to `wiki.json`
- Add Rainbow Bed to Rainbow Blocks Research Setup
- Add BED Material array in `ColoredMaterial.java`
- Add logic so both bed blocks update in `RainbowTickHandler.java`

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by svr333** [comment]
If anyone knows the issue with the bed switching sides, feel free to mention, otherwise Ill take a look at it tomorrow

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/handlers/RainbowTickHandler.java [hunk 1]
```diff
@@ -129,4 +157,38 @@ public boolean isSynchronized() {
...
+    private void setBedBlock(Block b, Material material, BlockFace facing, Part part, boolean doPhysics) {
+		b.setType(material, false);
+        BlockData data = b.getBlockData();
+        
+		if (data instanceof Bed) {
+			((Bed) data).setFacing(facing);
+            ((Bed) data).setPart(part);
+        }
+        
+        b.setBlockData(data, doPhysics);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/handlers/RainbowTickHandler.java:191]: Indentation hell awaits you

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Also, the block info is likely only stored on one side of the bed, so destroying the other half probably leaves a ghost block, right?
That would obviously need to be fixed...

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/handlers/RainbowTickHandler.java (hunk 1 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/handlers/RainbowTickHandler.java:191]: I dont know where you found that code but it looks perfectly fine on my end

**Comment by svr333** [comment]
I see, I do currently lack the knowledge to do this.
Ill check the docs on beds later when I have time for it

**Comment by TheBusyBiscuit** [comment]
I'm gonna close this for now.
If you find a way to address this issue, feel free to get back to us.
But I cannot really think of a "good" way to realize this concept at the moment either.

**Pull request closed without merging**